bin/pyenv-installer

```diff
@@ -37,18 +37,41 @@ if ! command -v git 1>/dev/null 2>&1; then
   exit 1
 fi
 
-if [ -n "${USE_GIT_URI}" ]; then
-  GITHUB="git://github.com"
+# Check ssh authentication if USE_SSH_URL is present
+if [ -n "${USE_SSH_URL}" ]; then
+  if ! command -v ssh 1>/dev/null 2>&1; then
+    echo "pyenv: configuration USE_SSH found but ssh is not installed, can't continue." >&2
+    exit 1
+  fi
+
+  ssh -T git@github.com 1>/dev/null 2>&1 || EXIT_CODE=$?
+  if [[ ${EXIT_CODE} != 1 ]]; then
+    {
+      echo "pyenv: github ssh authentication failed."
+      echo
+      echo "Considering generate a ssh key using ssh-keygen and register it first."
+      echo
+      echo "See the link below for more information:"
+      echo
+      echo "  https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors#check-your-ssh-access"
```

Comment:
Needs a rephrase to be more succinct and to the point.
Nothing to "consider", it's a strict requirement if the user wants to use SSH.

Reply:
99f5a9469c5f4e01adcf4175fc4efe4e419a967f